### PR TITLE
CI Global Reports

### PR DESCRIPTION
**Pull request by catlauraherzog**
Implemented global report scripts for all the report scripts like PHP Unit and PHP CS Fixer. In future there will be a lot of reports implemented through other tools like dicto. So we are going to place every report in one folder which the run.sh-script from CI-Results will grab. This PR is the main iteration for that.

### CODE REVIEW

**Comment by klees** [comment]
@lauraquellmalz There seems to be something wrong, the build fails.

**Comment by catlauraherzog** [comment]
@klees Hmm. You are right. The problem is the authentication to the CI-Results repo. But I didn't change the keys. And w/out acces to this repo I can't resolve it. @gvollbach do you have an idea?

File: CI/PHPUnit/run_tests_with_reporting.sh [hunk 1]
```diff
@@ -54,37 +54,25 @@ if [[ -e "$PHPUNIT_RESULTS_PATH" ]]
 				FAILURE=true
 		fi
 
-		if [[ "$TRAVIS_EVENT_TYPE" != "pull_request" ]]
+		if [[ "$TRAVIS_EVENT_TYPE" == "pull_request" ]]
```
Comment (gvollbach) [inline CI/PHPUnit/run_tests_with_reporting.sh]: @lauraquellmalz 
Why did you change the logic of this if statement? The results should only be committed if this is **NOT** a pull request.

**Comment by gvollbach** [comment]
@lauraquellmalz 
Good question, there is missing the repository key for the authentication, so you can not commit to the results repo. Why the key is missing, I don't have a clue at the moment, since you did not change anything in the results repo.
Could be because this is a pull request and so you are not the correct user for the key.

File: CI/PHPUnit/run_tests_with_reporting.sh (hunk 1 above)
Comment (catlauraherzog) [reply CI/PHPUnit/run_tests_with_reporting.sh]: Ups, sorry. Fixed.

**Comment by catlauraherzog** [comment]
> @lauraquellmalz
> Good question, there is missing the repository key for the authentication, so you can not commit to the results repo. Why the key is missing, I don't have a clue at the moment, since you did not change anything in the results repo.

Could it be that the key somehow deprecated? In my local setup with my forked repositories everything is fine.

**Comment by gvollbach** [comment]
@lauraquellmalz 
No that should not be the case. The key still works an normal commits, so I think this was a problem with the "pull request" and the "switched" condition.

**Comment by catlauraherzog** [comment]
@gvollbach
I think this change doesn't affect the run.sh script itself because this is used by report-results.sh. Anyway the build still bails due to authentication problems.

> remote: No anonymous write access.
> fatal: Authentication failed for 'https://@github.com/ILIAS-eLearning/CI-Results/'

It's interesting that this happens because I didn't touch the .travis.yml at all, just to add the report line. But that should not invoke the auth key. Weird.

**Comment by klees** [comment]
The problem might just be that PRs are running with a different user somehow. I mean, if one could use official ILIAS credentials in personal PRs for ILIAS this would open up the CI system for interesting injection attacks, right?

**Comment by catlauraherzog** [comment]
@klees Good point. Dunno if we can solve this.

**Comment by gvollbach** [comment]
Yes, thats why we didn't want the pull request, commit to the results repo.
@lauraquellmalz 
Yes thats the problem, you still trying to commit results, since you always call the run.sh script. The code before didn't call the run.sh script if this was a pull request. It only checks if there where errors and then exits.

**Comment by catlauraherzog** [comment]
@gvollbach So I should implement a condition to only report if it's not a PR?

**Comment by gvollbach** [comment]
@lauraquellmalz 
Would be the simplest way I think.

File: CI/results/report-results.sh [hunk 2]
```diff
@@ -1,5 +1,11 @@
 #!/bin/bash
 
+# Exit positively if we are on a PR
+if [[ "$TRAVIS_EVENT_TYPE" == "pull_request" ]];
+then
+  exit 0
```
Comment (gvollbach) [inline CI/results/report-results.sh]: @lauraquellmalz 
I think this would result in a always passed result, even when the unittests fail, that's why we have this construct here: https://github.com/ILIAS-eLearning/ILIAS/blob/release_6/CI/PHPUnit/run_tests_with_reporting.sh#L88 so if the unittest fail, we need another exit code, instead of 0. So the problem is not that simple. ;)

Comment (catlauraherzog) [reply CI/results/report-results.sh]: Oooooooh. I see ...

**Comment by catlauraherzog** [comment]
@gvollbach After some chatting with @klees with some more information I've changed the behavior so that the script only starts if the travis event is a cron job. Is that okay then?

**Comment by gvollbach** [comment]
@lauraquellmalz 
That would imply that we don't, get results for the normal commits and can't use the dashboard anymore, the cron only runs one time a week.

**Comment by catlauraherzog** [comment]
> @lauraquellmalz
> That would imply that we don't, get results for the normal commits and can't use the dashboard anymore, the cron only runs one time a week.

Then we should call again to determine how we should do it best way :-)

**Comment by gvollbach** [comment]
@lauraquellmalz 
Looks good to me! Thx for your work.

**Review by gvollbach — APPROVED** [review]

**Comment by klees** [comment]
Hi @gvollbach,

we could just merge this, right?

Best regards!

**Comment by gvollbach** [comment]
@klees Sure.
